tests/integration/test_data_store_mgr.py

```diff
@@ -0,0 +1,160 @@
+    flow_msg = schd.data_store_mgr.get_entire_workflow()
+    assert len(flow_msg.task_proxies) == len(data[TASK_PROXIES])
+
+
+def test_increment_graph_elements(harness):
+    """Test method that adds and removes elements by cycle point."""
+    schd, data = harness
+    assert schd.data_store_mgr.pool_points
+    assert len(data[TASK_PROXIES]) == 1
+
+
+def test_initiate_data_model(harness):
+    """Test method that generates all data elements in order."""
+    schd, data = harness
+    assert len(data[WORKFLOW].task_proxies) == 1
+    schd.data_store_mgr.initiate_data_model(reloaded=True)
+    assert len(data[WORKFLOW].task_proxies) == 1
+
+
+def test_prune_points(harness):
+    """Test method that removes data elements by cycle point."""
+    schd, data = harness
+    points = schd.data_store_mgr.cycle_states.keys()
+    point = next(iter(points))
+    assert point in points
+    schd.data_store_mgr.clear_deltas()
+    schd.data_store_mgr.prune_points([point])
+    schd.data_store_mgr.apply_deltas()
+    assert point not in points
+
+
+def test_update_data_structure(harness):
+    """Test update_data_structure. This method will generate and
+    apply deltas/updates given."""
+    schd, data = harness
+    assert len(collect_states(data, TASK_PROXIES)) == 1
+    update_tasks = schd.pool.get_all_tasks()
+    schd.data_store_mgr.update_data_structure(update_tasks)
+    assert len(update_tasks) > 0
+    assert len(update_tasks) == len(collect_states(data, TASK_PROXIES))
+
+
+def test_update_family_proxies(harness):
+    """Test update_family_proxies. This method will update all
+    DataStoreMgr task_proxies of given cycle point strings."""
+    schd, data = harness
+    assert len(collect_states(data, FAMILY_PROXIES)) == 1
+    update_tasks = schd.pool.get_all_tasks()
+    update_points = set((str(t.point) for t in update_tasks))
+    schd.data_store_mgr.clear_deltas()
+    schd.data_store_mgr.update_task_proxies(update_tasks)
+    schd.data_store_mgr.update_family_proxies(update_points)
+    schd.data_store_mgr.apply_deltas()
+    # Find families in updated cycle points
+    point_fams = [
+        f.id
+        for f in data[FAMILY_PROXIES].values()
+        if f.cycle_point in update_points
+    ]
+    assert len(point_fams) > 0
+    assert len(point_fams) == len(collect_states(data, FAMILY_PROXIES))
+
+
+def test_update_task_proxies(harness):
+    """Test update_task_proxies. This method will iterate over given
+    task instances (TaskProxy), and update any corresponding
+    DataStoreMgr task_proxies."""
+    schd, data = harness
+    assert len(collect_states(data, TASK_PROXIES)) == 1
+    update_tasks = schd.pool.get_all_tasks()
+    schd.data_store_mgr.clear_deltas()
+    schd.data_store_mgr.update_task_proxies(update_tasks)
+    schd.data_store_mgr.apply_deltas()
+    assert len(update_tasks) > 0
+    assert len(update_tasks) == len(collect_states(data, TASK_PROXIES))
+
+
+@pytest.mark.skip('TODO: fix this test')
```

Comment:
The update doesn't do anything because the workflow hasn't changed state, how do I trick it?